QuasarHelios2015/src/org/team1540/quasarhelios/AutonomousModeBase.java

```diff
@@ -0,0 +1,62 @@
+package org.team1540.quasarhelios;
+
+import ccre.channel.BooleanInputPoll;
+import ccre.channel.FloatInputPoll;
+import ccre.ctrl.BooleanMixing;
+import ccre.ctrl.FloatMixing;
+import ccre.holders.TuningContext;
+import ccre.instinct.AutonomousModeOverException;
+import ccre.instinct.InstinctModeModule;
+
+public abstract class AutonomousModeBase extends InstinctModeModule {
+	private FloatInputPoll encoderScaling;
+	public AutonomousModeBase(String modeName) {
+		super(modeName);
+	}
+	
+	protected void drive(float distance) throws AutonomousModeOverException, InterruptedException {
+		float currentEncoder = DriveCode.leftEncoder.get() * encoderScaling.get();
+		if (distance > 0) {
+			DriveCode.allMotors.set(1.0f);
+		} else {
+			DriveCode.allMotors.set(-1.0f);
+		}
+
+		waitUntilAtLeast(FloatMixing.multiplication.of(DriveCode.leftEncoder, encoderScaling), currentEncoder * encoderScaling.get() + distance);
```

Comment:
Also, perhaps you want to switch between waitUntilAtLeast and waitUntilAtMost based on the direction that you're going.